torchdynamo/variables/torch.py

```diff
@@ -398,3 +401,119 @@ def handle_ntuple(value):
             return variables.LambdaVariable(handle_ntuple, **options)
         else:
             return handle_ntuple(args[0])
+
+
+class TorchPyOperator(VariableTracker):
+    def __init__(self, value, **kwargs):
+        super(TorchPyOperator, self).__init__(**kwargs)
+        self.value = value
+
+    def call_function(
+        self, tx, args: "List[VariableTracker]", kwargs: "Dict[str, VariableTracker]"
+    ) -> "VariableTracker":
+        from torch.utils._python_dispatch import TorchDispatchMode
+        from torch.utils._python_dispatch import enable_torch_dispatch_mode
+
+        from . import ListVariable
+        from . import TensorVariable
+        from . import UserFunctionVariable
+
+        assert kwargs is None or len(kwargs) == 0, "kwargs are not supported, yet"
+
+        def unwrap_real(arg):
+            if isinstance(arg, TensorVariable):
+                return arg.as_proxy().node.meta["example_value"]
+            if isinstance(arg, UserFunctionVariable):
+                return arg.fn
+            if arg.has_unpack_var_sequence(tx):
+                return [
+                    unwrap_real(arg_inner) for arg_inner in arg.unpack_var_sequence(tx)
+                ]
+            return arg
+
+        # Get values
+        u_args = [unwrap_real(arg) for arg in args]
+
+        def unwrap_proxy(arg):
+            try:
+                return arg.as_proxy()
+            except NotImplementedError:
+                return arg
+
+        def register_as_subgraph(fn, name, args):
+            import torchdynamo
+
+            gm, guards = torchdynamo.export(fn, *args)
+
+            next_name = None
+            i = 0
+            while not next_name:
+                candidate = f"name_{i}"
+                if candidate in tx.output.nn_modules:
+                    i += 1
+                else:
+                    next_name = candidate
+
+            gm.__name__ = next_name
+            src = LocalSource(gm)
```

Comment:
LocalSource should take the name of a local variable in the root frame dynamo is tracing.